extensions/amp-ad/0.1/amp-ad-api-handler.js

```diff
@@ -120,21 +116,52 @@ export class AmpAdApiHandler {
           }
         }, this.is3p_, this.is3p_));
 
+    // Install API that listen to ad response
+    if (waitForRenderStart.indexOf(this.baseInstance_.adType) >= 0) {
+      this.isSupportRenderStart_ = true;
+    }
+
+    if (this.isSupportRenderStart_) {
+      // If support render-start, create a race between render-start no-content
+      this.adResponsePromise_ = listenForMessagesOncePromise(this.iframe_,
+        ['render-start', 'no-content'], this.is3p_).then(value => {
+          if (value == 'render-start') {
+              //report performance
+          } else {
+            if (this.noContentCallback_) {
```

Comment:
I think we should always have `noContentCallback_()`

You don't need to address it in this PR, maybe add a TODO for it.